go/staking/tests/tester.go

```diff
@@ -26,11 +26,62 @@ import (
 
 const recvTimeout = 5 * time.Second
 
+// stakingTestsState holds the current state of staking tests.
+type stakingTestsState struct {
+	totalSupply *quantity.Quantity
+	commonPool  *quantity.Quantity
+
+	srcAccountGeneralBalance      quantity.Quantity
+	srcAccountNonce               uint64
+	srcAccountEscrowActiveBalance quantity.Quantity
+	srcAccountEscrowActiveShares  quantity.Quantity
+
+	destAccountGeneralBalance quantity.Quantity
+}
+
+func (s *stakingTestsState) update(backend api.Backend, consensus consensusAPI.Backend) {
+	totalSupply, err := backend.TotalSupply(context.Background(), consensusAPI.HeightLatest)
+	if err != nil {
+		panic(err)
```

Comment:
Why should this panic?  make it take a `* testing.T`, and use assertions.

Reply:
I have no strong preferences. Ok, I'll make it use ordinary testing assertions.

Reply:
Done.